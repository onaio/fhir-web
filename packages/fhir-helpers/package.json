{
  "name": "@opensrp/fhir-helpers",
  "version": "0.0.1",
  "description": "",
  "main": "dist/index.js",
  "types": "dist/types",
  "files": [
    "dist"
  ],
  "publishConfig": {
    "access": "public"
  },
  "repository": "https://github.com/opensrp/web",
  "scripts": {
    "test": "run -T test $INIT_CWD --verbose",
    "tsc": "run -T tsc",
    "lint": "run -T eslint ./**/*.{js,jsx,ts,tsx}",
    "copy": "run -T copyfiles -u 1 \"./src/**/*.{css,html}\" \"./dist/\"",
    "build": "run tsc && run transpile && run copy",
    "transpile": "run -T babel src -d dist --root-mode upward --extensions .ts,.tsx --ignore '**/*.test.ts,**/*.test.tsx,**/tests,**/__tests__'"
  },
  "jest": {
    "automock": false,
    "setupFiles": [
      "../../setupTests"
    ]
  },
  "bugs": {
    "url": "https://github.com/opensrp/web/issues"
  },
  "dependencies": {
    "lodash": "^4.17.21"
  },
  "author": "OpenSRP Engineering",
  "license": "Apache-2.0"
}
