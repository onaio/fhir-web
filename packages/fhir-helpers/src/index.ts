export * from './constants/codeSystems';
export * from './constants/extensions';
export * from './constants/valueSetEnums';
export * from './constants/resourceTypes';
export * from './utils/utils';
