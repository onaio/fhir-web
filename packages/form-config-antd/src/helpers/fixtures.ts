/* eslint-disable @typescript-eslint/camelcase */
export const manifestRelease4 = {
  identifier: '1.0.30',
  json: '{"forms_version":"1.0.16","identifiers":["blood_screening.json"}',
  appId: 'org.smartregister.foo',
  appVersion: '1.2.18',
  createdAt: '2020-06-09T14:40:04.189+02:00',
  updatedAt: '2020-06-09T14:40:04.189+02:00',
};

export const fixManifestReleases = [
  {
    identifier: '1.0.11',
    json: '{"forms_version":"1.0.11","identifiers":["blood_screening.json"}',
    appId: 'org.smartregister.reveal',
    appVersion: '1.2.14',
    createdAt: '2020-06-08T15:00:38.413+02:00',
    updatedAt: '2020-06-08T15:00:38.413+02:00',
  },
  {
    identifier: '1.0.12',
    json: '{"forms_version":"1.0.12","identifiers":["blood_screening.json"}',
    appId: 'org.smartregister.reveal',
    appVersion: '1.2.14',
    createdAt: '2020-06-08T15:03:00.950+02:00',
    updatedAt: '2020-06-08T15:03:00.950+02:00',
  },
  {
    identifier: '1.0.16',
    json: '{"forms_version":"1.0.16","identifiers":["blood_screening.json"}',
    appId: 'org.smartregister.reveal',
    appVersion: '1.2.16',
    createdAt: '2020-06-09T14:40:04.189+02:00',
    updatedAt: '2020-06-09T14:40:04.189+02:00',
  },
];

export const manifestFile1 = {
  createdAt: 'Jun 19, 2020, 12:31:37 PM',
  form_relation: '',
  id: '52',
  identifier: 'test-form-1.json',
  isDraft: false,
  isJsonValidator: false,
  jursdiction: '',
  label: 'test form',
  module: '',
  version: '1.0.26',
};

export const manifestFile2 = {
  createdAt: 'Jun 19, 2020, 4:23:22 PM',
  form_relation: '',
  id: '53',
  identifier: 'reveal-test-file.json',
  isDraft: false,
  isJsonValidator: false,
  jursdiction: '',
  label: 'test publish',
  module: '',
  version: '1.0.27',
};

export const manifestFile3 = {
  createdAt: 'Jun 19, 2020, 6:23:22 PM',
  form_relation: '',
  id: '54',
  identifier: 'foo.json',
  isDraft: false,
  isJsonValidator: false,
  jursdiction: '',
  label: 'foo',
  module: 'baz',
  version: '1.0.28',
};

export const draftFile1 = {
  ...manifestFile1,
  isDraft: true,
  isJsonValidator: false,
};

export const draftFile2 = {
  ...manifestFile2,
  isDraft: true,
  isJsonValidator: false,
};

export const draftFile3 = {
  ...manifestFile3,
  isDraft: true,
  isJsonValidator: false,
};

export const fixManifestFiles = [manifestFile1, manifestFile2];

export const FixManifestDraftFiles = [
  {
    ...fixManifestFiles[0],
    isDraft: false,
  },
  {
    ...fixManifestFiles[1],
    isDraft: false,
  },
];

export const downloadFile = {
  clientForm: {
    json: JSON.stringify(fixManifestFiles[0]),
  },
};
