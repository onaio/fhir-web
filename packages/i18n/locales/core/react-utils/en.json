{
  "404": "",
  "Actions": "",
  "Class": "",
  "Date created": "",
  "Error": "",
  "Fetching next page": "",
  "Go back": "",
  "Go home": "",
  "ID": "",
  "Load more options": "",
  "No data": "",
  "No data available": "",
  "Search": "",
  "Showing {{recordsFetchedNum}}; {{remainingRecords}} more records.": "",
  "Something went wrong": "",
  "Sorry, the resource you requested for, does not exist": "",
  "Sorry, you are not authorized to access this page": "",
  "status": "",
  "Unable to load dropdown options.": "",
  "View details": ""
}
