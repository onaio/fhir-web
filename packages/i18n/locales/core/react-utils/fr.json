{
  "403": "",
  "404": "",
  "Actions": "",
  "Error": "",
  "Go back": "",
  "Go home": "",
  "Search": "",
  "Something went wrong": "",
  "Sorry, the resource you requested for, does not exist": "",
  "Sorry, you are not authorized to access this page": ""
}
