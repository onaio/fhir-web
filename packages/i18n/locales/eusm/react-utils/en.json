{
  "403": "",
  "404": "",
  "Actions": "",
  "Error": "",
  "Fetching next page": "Fetching next page",
  "Go back": "",
  "Go home": "",
  "Load more options": "Load more options",
  "No data": "No data",
  "Search": "",
  "Showing {{recordsFetchedNum}}; {{remainingRecords}} more records.": "Showing {{recordsFetchedNum}}; {{remainingRecords}} more records.",
  "Something went wrong": "",
  "Sorry, the resource you requested for, does not exist": "",
  "Sorry, you are not authorized to access this page": "",
  "Unable to load dropdown options.": "Unable to load dropdown options."
}
