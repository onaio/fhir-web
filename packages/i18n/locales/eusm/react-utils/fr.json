{
  "403": "403",
  "404": "404",
  "Actions": "Actions",
  "Error": "Erreur",
  "Go back": "Retour",
  "Go home": "Retour à l'accueil",
  "Search": "Recherche",
  "Session Expired": "Session expirée",
  "Something went wrong": "Il y a eu un problème",
  "Sorry, the resource you requested for, does not exist": "Désolé, la ressource que vous avez demandée n'existe pas",
  "Sorry, you are not authorized to access this page": "Désolé, vous n'êtes pas autorisé à accéder à cette page"
}
