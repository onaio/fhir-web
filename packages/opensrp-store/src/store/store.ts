import { connectReducer, getConnectedStore } from '@onaio/connected-reducer-registry';
import { gateKeeperReducer, gateKeeperReducerName } from '@onaio/gatekeeper';
import reducerRegistry, { combine, Registry } from '@onaio/redux-reducer-registry';
import session, { reducerName as sessionReducer } from '@onaio/session-reducer';
import { Reducer } from 'redux';
import { RouterState } from 'connected-react-router';
import { Dictionary } from '@onaio/utils';

/** declare globals interface */
declare global {
  interface Window {
    __PRELOADED_STATE__: Dictionary;
  }
}

/** Initial reducers in the reducer registry */
const defaultReducers: Registry = {
  router: connectReducer as Reducer<RouterState>,
};

// Grab the state from a global variable injected into the server-generated HTML
const preloadedState = window.__PRELOADED_STATE__ || {};
// Allow the passed state to be garbage-collected
delete window.__PRELOADED_STATE__;

/** Add users reducer to registry */
defaultReducers[sessionReducer] = session;
defaultReducers[gateKeeperReducerName] = gateKeeperReducer;

/** The initial store for the reveal web app */
export const store = getConnectedStore(defaultReducers, preloadedState);

/** Set listener to add reducers to store when registered */
reducerRegistry.setChangeListener((reducers) => {
  store.replaceReducer(combine(reducers));
});
