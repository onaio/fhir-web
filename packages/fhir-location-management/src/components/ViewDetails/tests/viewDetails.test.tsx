import React from 'react';
import { store } from '@opensrp/store';
import { authenticateUser } from '@onaio/session-reducer';
import { QueryClient, QueryClientProvider } from 'react-query';
import nock from 'nock';
import { render, cleanup, screen, waitFor } from '@testing-library/react';
import { Router } from 'react-router';
import { Provider } from 'react-redux';
import { locationResourceType } from '@opensrp/fhir-helpers';
import userEvent from '@testing-library/user-event';
import { centralProvinceInclude, rootFhirLocationInclude } from './fixtures';
import { centralProvinceLoc } from './fixtures';
import { ViewDetails } from '..';
import { createMemoryHistory } from 'history';
import { setConfig } from '@opensrp/pkg-config';
import { Switch, Route } from 'react-router';
import { RoleContext } from '@opensrp/rbac';
import { superUserRole } from '@opensrp/react-utils';

jest.mock('fhirclient', () => {
  return jest.requireActual('fhirclient/lib/entry/browser');
});

setConfig('projectCode', 'eusm');
jest.setTimeout(10000);
nock.disableNetConnect();

const props = {
  fhirBaseURL: 'http://test.server.org',
};
const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
      cacheTime: 0,
    },
  },
});
// eslint-disable-next-line @typescript-eslint/no-explicit-any
const AppWrapper = (props: any) => {
  return (
    <Provider store={store}>
      <QueryClientProvider client={queryClient}>
        <RoleContext.Provider value={superUserRole}>
          <Switch>
            <Route exact path={`/view/:id`}>
              <ViewDetails {...props} />
            </Route>
          </Switch>
        </RoleContext.Provider>
      </QueryClientProvider>
    </Provider>
  );
};

beforeAll(() => {
  store.dispatch(
    authenticateUser(
      true,
      {
        email: 'bob@example.com',
        name: 'Bobbie',
        username: 'RobertBaratheon',
      },
      { api_token: 'hunter2', oAuth2Data: { access_token: 'hunter2', state: 'abcde' } }
    )
  );
});

afterEach(() => {
  nock.cleanAll();
  cleanup();
});

afterAll(() => {
  nock.enableNetConnect();
});

test('renders correctly', async () => {
  const history = createMemoryHistory();
  history.push(`view/${centralProvinceLoc.id}`);

  nock(props.fhirBaseURL)
    .get(`/${locationResourceType}/_search`)
    .query({ _id: centralProvinceInclude.entry[0].resource.id, _include: 'Location:partof' })
    .reply(200, centralProvinceInclude);

  nock(props.fhirBaseURL)
    .get(`/${locationResourceType}/_search`)
    .query({ _id: rootFhirLocationInclude.entry[0].resource.id, _include: 'Location:partof' })
    .reply(200, rootFhirLocationInclude);

  render(
    <Router history={history}>
      <AppWrapper {...props}></AppWrapper>
    </Router>
  );

  // pageTitle - confirm page has loaded.
  await waitFor(() => {
    screen.getByText(/View details | Central ProvincegeByText/i);
  });
  // path breadcrumb
  const navLinks = document.querySelectorAll('nav ol li');
  for (const link of navLinks) {
    expect(link).toMatchSnapshot('breadcrumb path link');
  }

  // location details
  const detailSection = document.querySelector('[data-testid="details-section"]');
  expect(detailSection?.textContent).toEqual(
    'Central ProvinceEdit detailsID: d9d7aa7b-7488-48e7-bae8-d8ac5bd09334Version: 1Date Last Updated12/1/2023, 7:44:56 AMLocation NameCentral ProvinceStatusactivealiasLatitude & LongitudePhysical TypeJurisdictionGeometryAdministrative Level1Description'
  );

  // details tab is shown
  screen.getByTestId('details-tab');

  // edit location
  const editLocationBtn = screen.getByText(/Edit details/i);
  userEvent.click(editLocationBtn);
  expect(history.location.pathname).toEqual(
    '/admin/location/unit/edit/d9d7aa7b-7488-48e7-bae8-d8ac5bd09334'
  );

  expect(nock.isDone()).toBeTruthy();
});
