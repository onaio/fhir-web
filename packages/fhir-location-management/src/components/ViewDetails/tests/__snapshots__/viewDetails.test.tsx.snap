// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly: breadcrumb path link 1`] = `
<li>
  <span>
    Root FHIR Location
  </span>
</li>
`;

exports[`renders correctly: breadcrumb path link 2`] = `
<li
  aria-hidden="true"
  class="ant-breadcrumb-separator"
>
  &gt;
</li>
`;

exports[`renders correctly: breadcrumb path link 3`] = `
<li>
  <a
    href="/admin/location/unit/view/fe9e549b-a427-4db6-aad9-edade11b6e6a"
  >
    True Kenya
  </a>
</li>
`;

exports[`renders correctly: breadcrumb path link 4`] = `
<li
  aria-hidden="true"
  class="ant-breadcrumb-separator"
>
  &gt;
</li>
`;

exports[`renders correctly: breadcrumb path link 5`] = `
<li>
  <span>
    Central Province
  </span>
</li>
`;
